articles/feature-engineering-in-machine-learning/index.md

```diff
@@ -0,0 +1,167 @@
+
+- **Text features**. Text features are derived from text data. This means that text is encoded to corresponding numerical values.
+
+- **Image features**. Images are converted into suitable formats for analysis. The result is image-based features.
+
+Understanding of features puts us in a better position to dive into feature engineering techniques.  
+
+Let’s briefly define 5 techniques.
+
+##### 5 Feature Engineering Techniques
+###### Imputation
+It is common to deal with incomplete datasets. As we mentioned previously, feature engineering involves handling missing values. The technique responsible for this is imputation.
+
+###### Outliers
+Outliers are datapoints observed to be too far from the rest. They distort the results of the models therefore they must be handled. This technique deals with first identifying then trimming the outliers.
+
+###### Log Transform
+Skewness refers to the measure of how asymmetric a data distribution can be. When a dataset is skewed, it impacts performance of a model. Log transform comes in to fix skewness of a dataset. It has a purpose to make the dataset distribution as close to normal as possible. This technique also reduces the impact of outliers. It is worth noting that this technique works only with positive values.
+
+###### Binning
+Overfitting refers to a model that contains more parameters than can be accounted for by dataset. Noisy data contributes to overfitting. As a result, we use a binning technique to smooth noisy data. This involves dividing features into different bins. These features may be numerical or categorical.
+
+###### Feature Split
+This simply refers to splitting of a feature into two or more parts. This may be done to create new features or to aid an algorithm to get a better understanding of the dataset.
+
+#### Feature Engineering vs Feature Selection
+We have already defined and explored the feature engineering process. Nonetheless, it is common to have newbies use feature engineering and feature selection interchangeably.
+
+As we learned, feature engineering creates new features from raw data. However, feature selection is a process that involves selection of features of the highest influence on the target variable, from a set of existing features.
+
+This means the selection is a subset of existing features. The benefits of feature selection include:
+
+- Improving model accuracy when correct features are selected.
+
+- Reduction in [data dimensionality](https://www.statisticshowto.com/dimensionality/#:~:text=Dimensionality%20in%20statistics%20refers%20to,one%20column%20representing%20each%20dimension.). The result is reduced complexity of the model. This in turn means a model can be trained faster.
```

Comment:
Including the curse of dimensionality here and how and why ML models fail with higher dimensional data would be helpful.